### PR TITLE
Added an OpenCV binarization stage

### PR DESCRIPTION
**Pull request by Casper-NS**
more binarization with adaptive thresholding for comparison reasons

### CODE REVIEW

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/bitmap_process_stages/OpenCVBinarization.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.opencv.core.CvType
+import org.opencv.core.Mat
+import org.opencv.imgproc.Imgproc
+
+// Does not need grayscale as input.
+internal class OpenCVBinarization (
+    private val bitmap: Bitmap,
+    pipeline: Pipeline) : BitmapOutputStage(pipeline, Size(bitmap.width, bitmap.height), bitmap.config)
+    {
+        private var img = Mat(bitmap.width, bitmap.height, CvType.CV_8UC1)
```
Comment (mads256h) [inline app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/bitmap_process_stages/OpenCVBinarization.kt:17]: val

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/bitmap_process_stages/OpenCVBinarization.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        override fun update() {
+            Utils.bitmapToMat(bitmap, img)
+
+            var grayscale = Mat(img.size(), CvType.CV_8UC1)
+
+            Imgproc.cvtColor(img, grayscale, Imgproc.COLOR_RGB2GRAY, 4)
+
+            val binarized = Mat(img.size(), CvType.CV_8UC1)
+
+            Imgproc.adaptiveThreshold(grayscale, binarized,255.0, Imgproc.ADAPTIVE_THRESH_GAUSSIAN_C, Imgproc.THRESH_BINARY, 11, 2.0)
```
Comment (mads256h) [inline app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/bitmap_process_stages/OpenCVBinarization.kt:28]: What are these constants? Maybe make them customizable from the constructor

**Review by mads256h — CHANGES_REQUESTED** [review]

File: app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/bitmap_process_stages/OpenCVBinarization.kt (hunk 1 above)
Comment (Casper-NS) [reply app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/bitmap_process_stages/OpenCVBinarization.kt]: fixed

Comment (mads256h) [reply app/src/main/java/dk/scuffed/whiteboardapp/pipeline/stages/bitmap_process_stages/OpenCVBinarization.kt]: no it isnt

**Review by mads256h — APPROVED** [review]

**Pull request merged**